service/frontend/workflowHandler.go

```diff
@@ -2176,27 +1827,47 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		token.IsWorkflowRunning = isWorkflowRunning
 		token.PersistenceToken = nil
 	}
+	rawHistoryQueryEnabled := wh.config.EnableRawHistoryQuery() && wh.config.EnableRawHistoryQueryByDomain(getRequest.GetDomain())
 
 	history := &gen.History{}
+	historyBlob := []*gen.DataBlob{}
 	history.Events = []*gen.HistoryEvent{}
 	if isCloseEventOnly {
 		if !isWorkflowRunning {
-			history, _, err = wh.getHistory(
-				scope,
-				domainID,
-				*execution,
-				lastFirstEventID,
-				nextEventID,
-				getRequest.GetMaximumPageSize(),
-				nil,
-				token.TransientDecision,
-				token.BranchToken,
-			)
-			if err != nil {
-				return nil, wh.error(err, scope)
+			if rawHistoryQueryEnabled {
+				historyBlob, _, err = wh.getRawHistory(
```

Comment:
I didn't see the tests being updated to exercise this new code path. Can you port the rawHistory tests (which are deleted) to exercise this code instead ?

Reply:
> I didn't see the tests being updated to exercise this new code path. Can you port the rawHistory tests (which are deleted) to exercise this code instead ?

We did not have any unit test for GetRawHistoryExecution API on server side. Now I am trying to add integration test but still not find a way to turn on/off the knobs. Yesterday you mentioned we can add something to RPC header for that purpose, however, I am still investigating how to make it.

Reply:
the rpc header idea was for the client side integration test. For the server side integration test, we can tweak dynamic config to enable raw history for the test domain.

Reply:
@luowenking1 you can use the previous integration test that I wrote for GetWorkflowRawHistory and PollForWorkflowExecutionHistory.

Reply:
> @luowenking1 you can reuse the previous integration test that I wrote for GetWorkflowRawHistory and PollForWorkflowExecutionHistory.

Yes I have added the tests in the gethistory_test.go file.